### DISCUSSION TITLE
Aligner memory limit vs Clustering memory limit ?

### DISCUSSION THREAD

**Discussion by Sherman-1** [bbuchfink/diamond#764] [Q&A]
Hi,

I'm currently running Diamond 2.0 blastp for 500.000 queries against the last NR I could find ( 600Millions index ). I encountered a behavior that I don't understand. 

When running my blastp command with --block-size 10, I had a warning issue in my log file of this shape : 
"Warning: resident size (" << (res_size + last_size) << ") exceeds memory limit."

As I understand it, this warning is triggered when the following condition is met : 
if (res_size + last_size > mem_limit)

However, I find intriguing that the mem_limit constant is defined in the align_queries object as following :
const int64_t mem_limit = Util::String::interpret_number(config.memory_limit.get("16G")); 

Why is this variable is fixed at 16GB for the user ? I also saw in the config.cpp file that there is a memory_limit variable that can be modified but only for clustering commands. Are these two limits related in any way ? 

Last remark, when I run the exact same blastp run but with --block-size 8, the warning is not triggered anymore. 
Both runs have 200GB RAM available while my PBS executor estimate a peak consumption of 130GB : 

Job Id: 564147.pbsserver
    Job_Name = diamd_smple
    Job_Owner = simon.herman@node04.example.org
    resources_used.cpupercent = 4225
    resources_used.cput = 12:25:39
    resources_used.mem = 138869908kb
    resources_used.ncpus = 75
    resources_used.vmem = 45310404kb

**Comment by bbuchfink** [comment]
You can use the `-M` option to set a memory limit, but only for the clustering workflow. It does not apply to the alignment workflow and you can ignore that warning.

**Reply by Sherman-1** [reply]
Thank you very much !
